test/spec/reference-token-spec.js

```diff
@@ -0,0 +1,87 @@
+const SimpleAuthorization = artifacts.require('SimpleAuthorization');
+const ReferenceToken = artifacts.require('ReferenceToken');
+
+async function failTransaction(func, args, errorMessage) {
+  try {
+    await func.apply(this, args);
+    throw "Should have failed";
+  } catch (e) {
+    assert.equal(e.message, errorMessage);
+  }
+}
+
+revertMessage = "VM Exception while processing transaction: revert";
+
+contract('ReferenceToken', (accounts) => {
+  let simpleAuthorization;
+  let referenceToken;
+  const name = "testToken";
+  const symbol = "TKN";
+  const granularity = 16;
+
+
+  beforeEach(async () => {
+    simpleAuthorization = await SimpleAuthorization.new()
+    referenceToken = await ReferenceToken.new(name, symbol, granularity, simpleAuthorization.address);
+  });
+
+  it('should get instance of reference token', () => {
+    assert.isNotNull(referenceToken);
+  });
+
+  it('authorized should get tokens', async () => {
+      targetAccount = accounts[1];
+      const amountToMint = 100;
+
+      await simpleAuthorization.setAuthorized(targetAccount, true);
+      mintResult = await referenceToken.mint(targetAccount, amountToMint * granularity);
+      validationEvent = mintResult.logs[0];
+      assert.equal(validationEvent.event, 'Validation');
+      assert.equal(validationEvent.args.user, targetAccount);
+
+      balance = await referenceToken.balanceOf(targetAccount);
+      assert.equal(balance, amountToMint * granularity);
+  });
+
+  it('reference token receiver (mint) should be authorized', async () => {
+      targetAccount = accounts[1];
+      const amountToMint = 100;
+      await failTransaction(referenceToken.mint, [targetAccount, amountToMint * granularity], revertMessage);
+      await simpleAuthorization.setAuthorized(targetAccount, true);
+      await referenceToken.mint(targetAccount, amountToMint * granularity);
```

Comment:
No `assert`? Or is the test that it blows up on failure?

Reply:
I'm not sure how can you do it with assert.

failTransaction is a wrapper for cases when EVM throws an error (revertMessage in this case helps to check that it is expected error)

Reply:
@naumenkogs apologies, this has an `assert` in it. I was looking all over in the docs for `failTransaction`, but it's defined at the top of the file 😅 It has an assert in it:

`assert.equal(e.message, errorMessage)` 👍🏻

Reply:
Would this be a helpful function to port to `web3-server-tools`? Think that we'll be using it elsewhere?